packages/core/src/Stores/client-store.js

```diff
@@ -829,16 +839,35 @@ export default class ClientStore extends BaseStore {
         }
 
         runInAction(() => (this.is_switching = true));
+
+        this.resetLocalStorageValues(this.switched);
+        SocketCache.clear();
+
+        // if real to virtual --> switch to blue
+        // if virtual to real --> switch to green
+        // else keep the existing connection
+        const should_switch_socket_connection = this.is_virtual || /VRTC/.test(this.previous_login_id);
```

Comment:
How are we checking the if comments above? Seems like this is only checking for `is_virtual` state of the current account.

Reply:
Thanks i'll make this a bit simpler. It works currently since it also compares with the previous login_id but yeah it's difficult to read.

Reply:
@reza-fs updated it a bit.

If the account you are switching `to` or `from` is virtual it means that you are going from virtual <-> real since there can only be one virtual account.